ts/utils/providers/provider_factory.ts

```diff
@@ -1,72 +1,72 @@
+//         const injectedProviderIfExists = (window as any).ethereum;
+//         if (injectedProviderIfExists !== undefined) {
+//             return injectedProviderIfExists;
+//             // const provider = providerUtils.standardizeOrThrow(injectedProviderIfExists);
+//             // return provider;
+//         }
+//         const injectedWeb3IfExists = (window as any).web3;
+//         if (injectedWeb3IfExists !== undefined && injectedWeb3IfExists.currentProvider !== undefined) {
+//             const currentProvider = injectedWeb3IfExists.currentProvider;
+//             const provider = providerUtils.standardizeOrThrow(currentProvider);
+//             return provider;
+//         }
+//         return undefined;
+//     },
 
-    getWalletLinkProvider: (network: Network): ZeroExProvider => {
-        // NOTE: chainId will not update depending on the user's setting in the Coinbase Wallet app like with MetaMask
-        const rpcUrl = PUBLIC_NODE_URLS_BY_NETWORK_ID[network][0];
-        const walletLinkProvider = walletLink.makeWeb3Provider(rpcUrl, network);
+//     getWalletLinkProvider: (network: Network): ZeroExProvider => {
+//         // NOTE: chainId will not update depending on the user's setting in the Coinbase Wallet app like with MetaMask
+//         const rpcUrl = PUBLIC_NODE_URLS_BY_NETWORK_ID[network][0];
+//         const walletLinkProvider = walletLink.makeWeb3Provider(rpcUrl, network);
 
-        if (!walletLinkProvider || !walletLinkProvider.enable) {
-            throw new Error('Failed to connect to WalletLink');
-        }
+//         if (!walletLinkProvider || !walletLinkProvider.enable) {
+//             throw new Error('Failed to connect to WalletLink');
+//         }
 
-        // HACK(kimpers): TEMPORARY HACK until we revamp wallet handling
-        // Currently can't use standardizeOrThrow because it's checking this.accounts.length to
-        // determine if we are already connected which does not exist in WalletLink
-        const standardizedProvider: ZeroExProvider = {
-            isMetaMask: false,
-            isParity: false,
-            stop: _.noop.bind(_),
-            enable: (walletLinkProvider.enable as any).bind(walletLinkProvider),
-            sendAsync: (walletLinkProvider.sendAsync as any).bind(walletLinkProvider),
-        };
+//         // HACK(kimpers): TEMPORARY HACK until we revamp wallet handling
+//         // Currently can't use standardizeOrThrow because it's checking this.accounts.length to
+//         // determine if we are already connected which does not exist in WalletLink
+//         const standardizedProvider: ZeroExProvider = {
+//             isMetaMask: false,
+//             isParity: false,
+//             stop: _.noop.bind(_),
+//             enable: (walletLinkProvider.enable as any).bind(walletLinkProvider),
+//             sendAsync: (walletLinkProvider.sendAsync as any).bind(walletLinkProvider),
+//         };
```

Comment:
I think you need to do this hack for the WalletLink and WalletConnect web3-react providers before they can be passed into the staking contract wrappers. Same reason as above.